osc-server/src/main/java/org/osc/core/broker/service/securitygroup/ListSecurityGroupBindingsBySgService.java

```diff
@@ -62,14 +63,17 @@
         Set<VirtualSystem> vsSet = sg.getVirtualizationConnector().getVirtualSystems();
         long order = -1;
 
-        //Existing bindings
+        // Existing bindings
         for (SecurityGroupInterface sgInterface : sg.getSecurityGroupInterfaces()) {
             VirtualSystem vs = sgInterface.getVirtualSystem();
             vsSet.remove(vs);
-            VirtualSystemPolicyBindingDto virtualSystemBindingDto = new VirtualSystemPolicyBindingDto(vs.getId(), vs
-                    .getDistributedAppliance().getName(), sgInterface.getMgrPolicy() == null ? null : sgInterface.getMgrPolicy().getId(),
-                            FailurePolicyType.valueOf(sgInterface.getFailurePolicyType().name()),
-                            sgInterface.getOrder());
+            Set<Long> policyIds = new HashSet<>();
+            for (Policy policy : sgInterface.getMgrPolicies()) {
```

Comment:
You can retrieve the fields in a collection in Java 8 using Streams. I.e.:

``` 
List<String> ids = 
    entityList.stream()
              .map(Entity::getId)
              .collect(Collectors.toList());```

Reply:
Also look for other places in your code this might apply

Reply:
Done